transform/transformer_test.go

```diff
@@ -104,24 +105,65 @@ func TestLogTransformer_AddHeader(t *testing.T) {
 			transformer: LogTransformer{
 				LogContent: "+[+] helloworld",
 				TagID:      "some github diff",
-				Vcs: "github",
+				Vcs:        "github",
 			},
 			expected: "\n## cdk diff for some github diff \n<details>\n<summary>Click to expand</summary>\n\n```diff\n+[+] helloworld\n```\n</details>\n",
 		},
 		// when using vcs github but setting disable-collapse it should have no collapsible section
 		{
 			transformer: LogTransformer{
-				LogContent: "+[+] helloworld",
-				TagID:      "some github diff",
-				Vcs: "github",
+				LogContent:      "+[+] helloworld",
+				TagID:           "some github diff",
+				Vcs:             "github",
 				DisableCollapse: true,
 			},
 			expected: "\n## cdk diff for some github diff \n\n```diff\n+[+] helloworld\n```\n",
 		},
+		// when displaying overview for number of differences
+		{
+			transformer: LogTransformer{
+				LogContent:      "Number of stacks with differences: 2",
+				TagID:           "some github diff",
+				Vcs:             "github",
+				DisableCollapse: true,
+				ShowOverview:    true,
+			},
+			expected: "\n## cdk diff for some github diff \n\n\n```diff\nNumber of stacks with differences: 2\n```\n",
+		},
+		// when displaying overview for number of replaces
+		// {
+		// 	transformer: LogTransformer{
```

Comment:
I have debugged the test case. In test setup function we also require to call transform diff. Let's create a new test function for this purpose. Here is one example for one test 
```go
func TestLogTransformer_TransformDiffAddHeader(t *testing.T) {
	cases := []TemplateTest{
		//when displaying overview for number of replaces
		{
			transformer: LogTransformer{
				LogContent:      "[~] AWS::DynamoDB::Table ddb-table ddbtable7F3F6F3F replace\n └─ [~] TableName (requires replacement)\n-    ├─ [-] ddb-second-table\n+    └─ [+] ddb-second-table2",
				TagID:           "some github diff",
				Vcs:             "github",
				DisableCollapse: true,
				ShowOverview:    true,
			},
			contains: "⚠️  Number of resources that require replacement: 1",
		},
	}
	for _, c := range cases {
		c.transformer.transformDiff()
		c.transformer.addHeader()
		assert.Contains(t, c.transformer.LogContent, c.contains)
	}
}
```

Reply:
new test created